janusgraph-backend-testutils/src/main/java/org/janusgraph/graphdb/JanusGraphIndexTest.java

```diff
@@ -1002,33 +1067,51 @@ public void testIndexSelectStrategy() {
         mixedIndex.name();
         finishSchema();
 
+        final String backend = readConfig.get(INDEX_BACKEND, INDEX);
+        String nameKey, propKey;
+        if (backend.equals("solr")) {
```

Comment:
Nit: Ideally, `JanusGraphIndexTest` should be entirely independent of backend implementations. Exceptions for certain backends should be implemented in the corresponding index backend test (e.g. by overriding the original test case)

Reply:
Copying and pasting the whole test case does not sound elegant. Do you have advice on what helper method we could extract and override?

Reply:
You don't need to copy and paste the whole method because `JanusGraphIndexTest`.
You can create a method (maybe possible `abstract` but not necessary) which returns the exact piece of information which you need for specific index backend. 
One way is to create some methods like `getNameKey()` and `getPropertyKey()` which return `"name"` and `"prop"` by default but then overwrite that method in `solr`